src/wallet/wallet.cpp

```diff
@@ -2501,6 +2503,15 @@ bool CWallet::FundTransaction(CMutableTransaction& tx, CAmount& nFeeRet, int& nC
     return true;
 }
 
+static CFeeRate GetDiscardRate(const CBlockPolicyEstimator& estimator)
+{
+    unsigned int highest_target = estimator.HighestTargetTracked(FeeEstimateHorizon::LONG_HALFLIFE);
+    CFeeRate discard_rate = estimator.estimateSmartFee(highest_target, nullptr /* FeeCalculation */, false /* conservative */);
+    discard_rate = std::min(discard_rate, CWallet::m_discard_rate);
+    discard_rate = std::max(discard_rate, ::dustRelayFee);
```

Comment:
In commit "Add a discard_rate"

Maybe it should be an init error to specify `-dustrelayfee` greater than `-discardrate`, because the `-discardrate` value is ignored in this case.